.github/workflows/DiffTool-CI.yml

```diff
@@ -0,0 +1,136 @@
+name: CI/CD for DiffTool
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:  # Allows manual triggering
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    env:
+      JAVA_VERSION: '21'
+      CONFIG_DIR: '.ci-config'
+      DIFFTOOL_JAR: "diff-java-tool.jar"
+      PATCH_DIFF_TOOL_VERSION: "0.1-SNAPSHOT"
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          clean: true
+          fetch-depth: 1
```

Comment:
if you remove this line, it will be good git clone and you do not need hack `- name: Initialize git repository in 'extractor'`

Reply:
> if you remove this line, it will be good git clone and you do not need hack `- name: Initialize git repository in 'extractor'`

If we don't do it, it complains not a git repository. 
```
projects:
  - name: extractor
    scm: git
    url: file://./extractor
    reference: main
```
I get this below:
```
com.github.checkstyle.difftool.DiffTool - Cloning git repository 'extractor' to repositories/extractor folder ...
2024-10-03T00:52:33.4677094Z Cloning into 'repositories/extractor'...
2024-10-03T00:52:33.4722486Z fatal: '/extractor' does not appear to be a git repository
2024-10-03T00:52:33.4725850Z fatal: Could not read from remote repository.
```

Reply:
Extractor is subfolder.
Maybe we should point to root repository folder - level up / parent folder of extractor

Reply:
> Extractor is subfolder.
> Maybe we should point to root repository folder - level up / parent folder of extractor

Despite removing the fetch, it is still not working. The best explanation i could find from internet is below:

Reply:
Do you have any workflow in handy where we do that?